docs/book/src/developer/providers/contracts/infra-machinepool.md

```diff
@@ -0,0 +1,468 @@
+# Contract rules for InfraMachinePool
+
+Infrastructure providers CAN OPTIONALLY implement an InfraMachinePool resource.
+
+The goal of an InfraMachinePool is to manage the lifecycle of a provider-specific pool of machines using a provider specific service (like auto-scale groups in AWS & virtual machine scalesets in Azure).
+
+The machines in the pool may be physical or virtual instances (although most likely virtual), and they represent the infastructure for Kubernetes nodes.
```

Comment:
```suggestion
The machines in the pool may be physical or virtual instances (although most likely virtual), and they represent the infastructure for Kubernetes nodes.

The InfraMachinePool resource will be referenced by one of the Cluster API core resources, MachinePool.
```
for consistency with other contracts + it makes the bridge with the next sentence